### DISCUSSION TITLE
Add feature for creating iptables rules based on interface instead of IP

### DISCUSSION THREAD

**Discussion by dub452** [Enkidu-6/tor-ddos#74] [Ideas]
Hello,

Total noob here, I'm just requesting the feature to be able to input an interface like ppp0 instead of an IP address for those that run nodes on dynamic IPs. I know this is an edge case but I'd be stoked to be able to run this script for my node. I've been losing my flags and clogging my box for a few weeks now.

Thank you for your service!

**Comment by Enkidu-6** [comment]
Theoretically yes there is but since I'm not able to test it at the moment, I won't be able to give you a definitive answer and I can't be sure of side effects. But if you're willing to be the test case, I'll be glad to give you the instructions to do so. Do you also have an IPV6 or is it just a dynamic IPV4?

**Comment by Enkidu-6** [comment]
Just in case you're interested to try, do the following please:
- Enter your current  ipaddress:port in the correct format in ipv4.txt and ipv6.txt and remove the comment on top of the files.
- Edit multi.sh and put a # to the left of line 63 where it says ./rules.sh so it looks like #./rules.sh
- chmod 0700 multi.sh and run it by typing ./multi.sh. This will not apply any rules. It will only create a file named rules.sh so we can edit it later.
- Once you've done that, let me know so I can help you edit the file. We'll need your current IP address and the naming convention of your NIC to do the edit.
